hrm-service/src/contact/contact.ts

```diff
@@ -79,11 +112,31 @@ export const createContact = async (
       newContact.queueName || (<any>(rawJson ?? {})).queueName,
     createdBy,
   };
-  return create(
+
+  if (completeNewContact.taskId) {
```

Comment:
Yes, I think the reason was thinking on poor internet connections, where we have already seen "duplicated contacts" becuase of internet blips after the counselor requested to "save the form" but the server response got lost. We relied on Twilio generated task id to make a contact submission "truly unique". I'm not sure if this is something we want to keep or if we sould just get rid of it. Thoughts @nickhurlburt?

Reply:
I can totally see the logic of wanting to make creates idempotent and greacefully handle duplicate create requests - it's just more usual to keep the new version rather than the old one. At least then when you return a 200 telling the client it 'successfully created the contact', it's true, it did create the contact you asked it to - it just by overwriting another one. 

At least then code that assume a 200 means the contact will exist as they specified it, rather than, it will exist, but it could be different to what they specified.

Anyway, not something that needs to be addressed here

Reply:
My thoughts:
- As mentioned, we made creates idempotent so that we could safely retry from the frontend
- @stephenhand makes a valid point about what it means to return a 200 on a duplicate... I think previously we assumed that the use case was a retry of the exact same data, so there was no point in overwriting. Also, before we had a contact audit log, we wanted contacts to be immutable, further making the case for keeping the original. I suppose one thing we could do here is check if the existing contact 'deep equals' the new one, and return a 4XX error if doesn't deep equal (since then it's really an edit, not a create) and if it _is_ an exact copy, perhaps return a 204 No Content or a [409 Conflict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) -- we'd just have to make sure the client hands that appropriately. Either way, I might lean toward not worrying about that in this PR, but I leave that up to you all.
- @stephenhand 's other point about using transactions to avoid concurrency errors is a good one. In the past we have not been careful about transactions, but we should do better. I would lean toward keeping or adding protection against the kind of concurrency error he describes (how exactly, I leave to you all :) )